sanitycheck: fail on warnings

nashif opened the pull request.

nashif committed:
sanitycheck: fail on warnings

Missed adding extra cflags to fail on warning. Adding those back.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

galak commented on scripts/sanitycheck:
```diff
@@ -1777,8 +1777,16 @@ class CMake():
                       '-S{}'.format(self.source_dir),
+                      '-DEXTRA_CFLAGS="-Werror {}'.format(cflags),
+                      '-DEXTRA_AFLAGS=-Wa,--fatal-warnings',
+		      '-DEXTRA_LDFLAGS="{}'.format(ldflags),
```
Nit, feels like white space isn't aligned.

galak commented on scripts/sanitycheck:
```diff
@@ -1777,8 +1777,16 @@ class CMake():
 
         verbose("Running cmake on %s for %s" %(self.source_dir, self.platform.name))
 
+        ldflags = ""
```
seems like have aflags here as well.

galak commented on scripts/sanitycheck:
```diff
@@ -1777,9 +1777,17 @@ class CMake():
-        cmake_args = ['-B{}'.format(self.build_dir),
-                      '-S{}'.format(self.source_dir),
-                      '-G{}'.format(get_generator()[1])]
+        ldflags="-Wl,--fatal-warnings"
```
why is ldflags special?

nashif replied:
it is not, I copied the code as it was in previous version and modified it slightly while I look into a long term fix and addressing some other options we have to deal with errors and deprecations.

aescolar commented:
Fixes #20025

nashif committed:
ztest: add missing headers

Recent changes to architecture headers did not address ztest headers due
to this bug in sanitycheck. Fixing them now.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

aescolar commented:
Fixes #20049
